test/gtest/uct/ib/test_ib.cc

```diff
@@ -259,27 +320,54 @@ class test_uct_ib : public uct_test {
 protected:
     entity *m_e1, *m_e2;
     static size_t ib_am_handler_counter;
+    struct ibv_device **device_list;
+    int num_devices;
 };
 
 size_t test_uct_ib::ib_am_handler_counter = 0;
 
-UCS_TEST_P(test_uct_ib, non_default_pkey, "IB_PKEY=0x2")
+UCS_TEST_P(test_uct_ib, non_default_pkey)
 {
-    ib_port_desc_t *port_desc;
+    /* test with invalid pkey set */
+    for (unsigned pkey = 0; pkey <= UCT_IB_PKEY_PARTITION_MASK; pkey++) {
```

Comment:
how long does it take? Maybe test just a couple of keys?

Reply:
it exits the loop once  the first pkey that isn't configured is found
it is fast enough, because it will do `128` iterations in the bad case (when [0] = 1, [1] = 2, ... [127] = 128; pkey=129 will be invalid pkey and it exits the loop)